molo/core/migrations/0050_data_migration_promoted_articles.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+def set_promote_start_date(apps, schema_editor):
```

Comment:
This migration can be summed up as follows:
```python
from django.db.models import F, Q
ArticlePage = apps.get_model("core", "ArticlePage")
ArticlePage.objects.filter(
    Q(featured_in_latest=True) | Q(featured_in_homepage =True) | Q(featured_in_section =True)
).update(featured_in_latest_start_date=F('latest_revision_created_at'))
```
This does the exact same thing with 2 differences:
1. It avoids the loops. The SQL query ends up being 1 query rather than `n` updates.
2. `.update()` is run on a db level and so it'll be much faster than python.

Also.. you were missing a `.save()` in all those loops.